### PR TITLE
Implement conf-dict as Conf-class

### PR DESCRIPTION
**Pull request by phorward**
- Allows to perform deprecation behavior on config variables
- Implemented deprecation behavior for #552
- Remove unused config variables
- Fixed some deprecation warnings to become more clear (tested locally)
---
I'm assuming these defaults are unused, and propose a removal of them:

- `"viur.capabilities"` is used in renderer to provide some "additional" undocumented information about modules. Seems to be a relict of ViUR v1, never seen this in any recent project.
- `"viur.db.caching"` is not found in either viur-core or viur-datastore anymore
- The variable `apiVersion` was used NOWHERE.

### CODE REVIEW

**Review by ArneGudermann — CHANGES_REQUESTED** [review]
Hey @phorward. Nice feature, but I have some suggestions.

File: core/config.py [hunk 1]
```diff
@@ -1,18 +1,55 @@
...
+                raise ValueError(f"{item!r} was substituted by `viur.render.html.downloadUrlExpiration`, please fix!")
+
+        # Avoid to set conf values to something which is already the default
+        if item in self and self[item] == value:
+            msg = f"Setting conf[\"{item}\"] to {value!r} has no effect, as this value has already been set"
+            warnings.warn(msg)
+            logging.warning(msg)
+            return
+
+        super().__setitem__(item, value)
```
[the remark this answers is not in the document]
Comment (phorward) [reply core/config.py:40]: >  And why we should avoid to set an value to the same ? For this behavior we must check every time we insert something if its there and if it has the same value.

This is just to keep project's code clean. Sometimes people set config variables to values which are already the default. This informs the user that he/she has effect-less code which can be removed. I really would like to integrate this.

**Comment by phorward** [comment]
Hello @ArneGudermann, I've renamed the "item" variables into "key", as this is also the default in the [Python reference](https://docs.python.org/3/reference/datamodel.html#object.__getitem__). Nevertheless, I want to keep the warning when setting config variables to values which are already configured, as a little help to keep project settings clean, as described above in my comment.

**Review by ArneGudermann — APPROVED** [review]

**Comment by ArneGudermann** [comment]
Ok I'm fine with it can you please resolve the confilcts. Then we can merged it from my site

**Comment by ArneGudermann** [comment]
> Ok I'm fine with it can you please resolve the confilcts. Then we can merged it from my site

Never mind you already did it.

**Review by ArneGudermann — APPROVED** [review]

**Pull request merged**